### PR TITLE
docs: Add an example of type checking a middleware function

### PR DESCRIPTION
**Pull request by mszabo-wikia**
---
name: :book: New/Updated Documentation Content
about: Adding a new docs page, or updating content in an existing docs page
---

## PR Type

**Does this PR add a _new_ page, or update an _existing_ page?**
It updates an existing page.
## Checklist

- [ ] Is there an existing issue for this PR?
- [x] Have the files been linted and formatted?

## What docs page is being added or updated?

- **Section**: Recipes
- **Page**: Usage With TypeScript

### What updates should be made to the page?
It was pointed out in a Reactiflux conversation last night that the TypeScript
examples provided by Redux currently do not demonstrate how to type a middleware
function. This patch adds a brief section to the docs that shows how to define a
typed middleware.

### Do these updates change any of the assumptions or target audience? If so, how do they change?
This recipe assumes an entry-level knowledge of Redux and TypeScript and the new example aims to remain in line with these expectations.

### CODE REVIEW

File: docs/recipes/UsageWithTypescript.md [hunk 1]
```diff
@@ -230,6 +230,26 @@ const rootReducer = combineReducers({
...
+
+```ts
+// src/middleware/example.ts
+
+import { Middleware } from 'redux'
+
+import { RootState } from '../store'
+
+export const exampleMiddleware: Middleware<
+  {},
```
Comment (timdorr) [inline docs/recipes/UsageWithTypescript.md:246]: We may want to briefly explain what this generic is for, as it's not that obvious from this example.

**Review by timdorr — APPROVED** [review]
This looks good as-is. If want to address the Dispatch extension generic, we can, but I don't think that's a blocker.

File: docs/recipes/UsageWithTypescript.md (hunk 1 above)
Comment (mszabo-wikia) [reply docs/recipes/UsageWithTypescript.md:246]: Sure, I can add a note. Would you prefer it in a standalone commit or squashed?

Comment (timdorr) [reply docs/recipes/UsageWithTypescript.md:246]: Either is fine. We're going to squash it all anyways when it gets merged.

Comment (mszabo-wikia) [reply docs/recipes/UsageWithTypescript.md]: 👍  3c0023ac

**Comment by timdorr** [comment]
Thanks!

**Pull request merged**